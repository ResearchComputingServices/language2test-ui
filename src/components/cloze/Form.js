import React, { createRef } from 'react';
import PropTypes from 'prop-types';
import { FormPaper } from '../common';
import Form from '../form';

function ClozeForm({
    title,
    data,
    layout,
    buttons,
    controls,
    children,
}) {
    const ref = createRef();
    return (
        <FormPaper ref={ref}>
            <div className='form-body'>
                <h6 className='cloze-form-title'>
                    {title}
                </h6>
                <Form
                    buttons={buttons}
                    controls={controls}
                    data={data}
                    layout={layout}
                    sections={[{
                        layout: [{
                            type: 'raw',
                            content: <div className='field'>{children}</div>,
                        }],
                    }]}
                />
            </div>
        </FormPaper>
    );
}

ClozeForm.propTypes = {
    title: PropTypes.string,
    layout: PropTypes.array.isRequired,
    data: PropTypes.object,
    buttons: PropTypes.array,
    controls: PropTypes.object.isRequired,
    children: PropTypes.node.isRequired,
};

ClozeForm.defaultProps = {
    title: 'Cloze Question',
    data: {},
    buttons: [],
};

export default ClozeForm;
