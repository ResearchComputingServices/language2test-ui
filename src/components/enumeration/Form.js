import React from 'react';
import PropTypes from 'prop-types';
import MaterialTable from 'material-table';
import Add from '@material-ui/icons/Add';
import { FormPaper } from '../common';
import Form from '../form';

const enumerationValuesTableStyle = {
    width: '60vw',
    minWidth: 450,
    padding: 20,
};

function EnumerationForm({
    title,
    data,
    layout,
    values,
    onValueAdd,
    onValueUpdate,
    onValueDelete,
    buttons,
    controls,
}) {
    return (
        <FormPaper >
            <div className='form-body'>
                <h6 className='form-title'>
                    {title}
                </h6>
                <Form
                    buttons={buttons}
                    controls={controls}
                    data={data}
                    layout={layout}
                    sections={[{
                        layout: [{
                            type: 'raw',
                            content: (
                                <div className='d-flex flex-column justify-content-center align-items-center my-5'>
                                    <MaterialTable
                                        columns={[{ field: 'text' }]}
                                        data={values}
                                        editable={{
                                            onRowAdd: onValueAdd,
                                            onRowUpdate: onValueUpdate,
                                            onRowDelete: onValueDelete,
                                        }}
                                        icons={{
                                            Add: props => (
                                                <Add
                                                    {...props}
                                                    color='primary'
                                                />
                                            ),
                                        }}
                                        localization={{
                                            body: {
                                                addTooltip: 'Add Enumeration Value',
                                                editRow: { deleteText: 'Are you sure?' },
                                            },
                                        }}
                                        style={enumerationValuesTableStyle}
                                        title='Enumeration Values'
                                    />
                                </div>
                            ),
                        }],
                    }]}
                />
            </div>
        </FormPaper>
    );
}

EnumerationForm.propTypes = {
    title: PropTypes.string,
    data: PropTypes.object,
    values: PropTypes.array,
    layout: PropTypes.array.isRequired,
    controls: PropTypes.object.isRequired,
    buttons: PropTypes.array,
    onValueAdd: PropTypes.func,
    onValueUpdate: PropTypes.func,
    onValueDelete: PropTypes.func,
};

EnumerationForm.defaultProps = {
    title: 'Enumeration Question',
    data: {},
    values: [],
    buttons: [],
    onValueAdd: undefined,
    onValueUpdate: undefined,
    onValueDelete: undefined,
};

export default EnumerationForm;
