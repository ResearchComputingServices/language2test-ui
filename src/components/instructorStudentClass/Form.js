import React from 'react';
import PropTypes from 'prop-types';
import { FormPaper, InUse } from '../common';
import Form from '../form';

function InstructorStudentClassForm({
    title,
    data,
    layout,
    buttons,
    controls,
}) {
    return (
        <FormPaper >
            <div className='form-body'>
                <h6 className='form-title'>
                    {title}
                </h6>
                <InUse
                    msg='Currently being used in one or more Test Assignation(s)'
                    show={data.immutable || data.unremovable}
                />
                <Form
                    buttons={buttons}
                    controls={controls}
                    data={data}
                    layout={layout}
                />
            </div>
        </FormPaper>
    );
}

InstructorStudentClassForm.propTypes = {
    title: PropTypes.string,
    data: PropTypes.object,
    layout: PropTypes.array.isRequired,
    controls: PropTypes.object.isRequired,
    buttons: PropTypes.array,
};

InstructorStudentClassForm.defaultProps = {
    title: 'Student Class',
    data: {},
    buttons: [],
};

export default InstructorStudentClassForm;
