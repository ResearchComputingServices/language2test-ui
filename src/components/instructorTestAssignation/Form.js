import React from 'react';
import PropTypes from 'prop-types';
import { FormPaper } from '../common';
import Form from '../form';

function InstructorTestAssignationForm({
    title,
    data,
    layout,
    buttons,
    controls,
}) {
    return (
        <FormPaper >
            <div className='form-body'>
                <h6 className='form-title'>
                    {title}
                </h6>
                <Form
                    buttons={buttons}
                    controls={controls}
                    data={data}
                    layout={layout}
                />
            </div>
        </FormPaper>
    );
}

InstructorTestAssignationForm.propTypes = {
    title: PropTypes.string,
    data: PropTypes.object,
    layout: PropTypes.array.isRequired,
    controls: PropTypes.object.isRequired,
    buttons: PropTypes.array,
};

InstructorTestAssignationForm.defaultProps = {
    title: 'Test Schedules',
    data: {},
    buttons: [],
};

export default InstructorTestAssignationForm;
