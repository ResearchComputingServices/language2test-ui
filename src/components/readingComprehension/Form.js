import React, { createRef } from 'react';
import PropTypes from 'prop-types';
import { FormPaper } from '../common';
import Form from '../form';

function ReadingComprehensionForm({
    title,
    data,
    layout,
    dynamicLayout,
    controls,
    buttons,
    children,
}) {
    const ref = createRef();
    return (
        <FormPaper ref={ref}>
            <div className='form-body'>
                <h6 className='form-title'>
                    {title}
                </h6>
                <Form
                    buttons={buttons}
                    controls={controls}
                    data={data}
                    layout={layout.concat(dynamicLayout)}
                    sections={[{
                        layout: [{
                            type: 'raw',
                            content: <div className='field'>{children}</div>,
                        }],
                    }]}
                />
            </div>
        </FormPaper>
    );
}

ReadingComprehensionForm.propTypes = {
    title: PropTypes.string,
    dynamicLayout: PropTypes.array.isRequired,
    layout: PropTypes.array.isRequired,
    data: PropTypes.object,
    buttons: PropTypes.array,
    controls: PropTypes.object.isRequired,
    children: PropTypes.node.isRequired,
};

ReadingComprehensionForm.defaultProps = {
    title: 'Reading Comprehension Question',
    data: {},
    buttons: [],
};

export default ReadingComprehensionForm;
