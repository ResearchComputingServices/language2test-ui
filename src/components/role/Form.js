import React from 'react';
import PropTypes from 'prop-types';
import { FormPaper } from '../common';
import Form from '../form';

function RoleForm({
    title,
    data,
    layout,
    buttons,
    controls,
    children,
}) {
    return (
        <FormPaper >
            <div className='form-body-wide'>
                <h6 className='form-title'>
                    {title}
                </h6>
                <Form
                    buttons={buttons}
                    controls={controls}
                    data={data}
                    layout={layout}
                    sections={[{
                        layout: [{
                            type: 'raw',
                            content: children,
                        }],
                    }]}
                />
            </div>
        </FormPaper>
    );
}

RoleForm.propTypes = {
    title: PropTypes.string,
    data: PropTypes.object,
    layout: PropTypes.array.isRequired,
    controls: PropTypes.object.isRequired,
    buttons: PropTypes.array,
    children: PropTypes.node,
};

RoleForm.defaultProps = {
    title: 'Role',
    data: {},
    buttons: [],
    children: null,
};

export default RoleForm;
