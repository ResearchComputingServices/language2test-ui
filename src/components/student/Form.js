import React from 'react';
import PropTypes from 'prop-types';
import { FormPaper, Button } from '../common';
import Form from '../form';
import { ApiPicklistField } from '../form/fields';

function StudentForm({
    title,
    data,
    layout,
    dynamicLayout,
    dynamicData,
    onAddDemographicField,
    disableAddDemographicField,
    selectedDemographicFieldsFilter,
    buttons,
    controls,
}) {
    const configureSections = () => {
        const sections = [
            {
                layout: dynamicLayout,
                data: dynamicData,
                append: (
                    <div className='d-flex justify-content-end align-items-center'>
                        <div>
                            <ApiPicklistField
                                controls={controls}
                                field={{
                                    name: 'selectedDemographicField',
                                    label: 'Demographic Fields',
                                    type: 'api-picklist-multiple',
                                    entity: 'demographicQuestionnaireField',
                                    filter: selectedDemographicFieldsFilter,
                                }}
                            />
                            <Button
                                className='mt-1 mr-2'
                                color='primary'
                                disabled={disableAddDemographicField}
                                onClick={onAddDemographicField}
                                size='small'
                                variant='contained'
                            >
                                Add Demographic Field
                            </Button>
                        </div>
                    </div>
                ),
            },
        ];
        return sections;
    };
    return (
        <FormPaper >
            <div className='form-body'>
                <h6 className='form-title'>
                    {title}
                </h6>
                <Form
                    buttons={buttons}
                    controls={controls}
                    data={data}
                    layout={layout}
                    sections={configureSections()}
                />
            </div>
        </FormPaper>
    );
}

StudentForm.propTypes = {
    title: PropTypes.string,
    data: PropTypes.object.isRequired,
    layout: PropTypes.array.isRequired,
    dynamicData: PropTypes.object.isRequired,
    dynamicLayout: PropTypes.array.isRequired,
    onAddDemographicField: PropTypes.func.isRequired,
    selectedDemographicFieldsFilter: PropTypes.func.isRequired,
    disableAddDemographicField: PropTypes.bool.isRequired,
    buttons: PropTypes.array,
    controls: PropTypes.object.isRequired,
};

StudentForm.defaultProps = {
    title: 'Student',
    buttons: [],
};

export default StudentForm;
