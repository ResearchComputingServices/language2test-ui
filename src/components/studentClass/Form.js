import React from 'react';
import PropTypes from 'prop-types';
import { FormPaper } from '../common';
import Form from '../form';

function StudentClassForm({
    title,
    data,
    layout,
    buttons,
    controls,
}) {
    return (
        <FormPaper >
            <div className='form-body'>
                <h6 className='form-title'>
                    {title}
                </h6>
                <Form
                    buttons={buttons}
                    controls={controls}
                    data={data}
                    layout={layout}
                />
            </div>
        </FormPaper>
    );
}

StudentClassForm.propTypes = {
    title: PropTypes.string,
    data: PropTypes.object,
    layout: PropTypes.array.isRequired,
    controls: PropTypes.object.isRequired,
    buttons: PropTypes.array,
};

StudentClassForm.defaultProps = {
    title: 'Student Class',
    data: {},
    buttons: [],
};

export default StudentClassForm;
