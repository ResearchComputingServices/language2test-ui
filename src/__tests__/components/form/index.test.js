import Form from '../../../components/form';
import sampleLayout from '../../../config/formLayouts/demographicQuestionnaireFormLayout';
import { createSnapshotTest } from '../../../testUtilities';

describe('form', () => {
    it('should match snapshot', createSnapshotTest(Form, {
        props: {
            initialValues: {},
            controls: {},
            layout: sampleLayout,
            onSubmit: jest.fn(),
            buttonTitle: 'Save',
            data: {},
        },
    }));
});
