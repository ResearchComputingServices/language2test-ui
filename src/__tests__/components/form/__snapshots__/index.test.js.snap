// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form should match snapshot 1`] = `
<div
  id="generated-form"
>
  <MuiPickersUtilsProvider
    utils={[Function]}
  >
    <Field
      controls={Object {}}
      field={
        Object {
          "defaultValue": undefined,
          "disabled": true,
          "field": "name",
          "fullWidth": true,
          "label": "Username",
          "name": "name",
          "required": true,
          "title": "Username",
          "type": "text",
          "variant": "outlined",
        }
      }
      key="1"
    >
      <WithStyles(ForwardRef(TextField)) />
    </Field>
    <Field
      controls={Object {}}
      field={
        Object {
          "defaultValue": undefined,
          "disabled": true,
          "field": "firstName",
          "fullWidth": true,
          "label": "First Name",
          "name": "firstName",
          "required": true,
          "title": "First Name",
          "type": "text",
          "variant": "outlined",
        }
      }
      key="2"
    >
      <WithStyles(ForwardRef(TextField)) />
    </Field>
    <Field
      controls={Object {}}
      field={
        Object {
          "defaultValue": undefined,
          "disabled": true,
          "field": "lastName",
          "fullWidth": true,
          "label": "Last Name",
          "name": "lastName",
          "required": true,
          "title": "Last Name",
          "type": "text",
          "variant": "outlined",
        }
      }
      key="3"
    >
      <WithStyles(ForwardRef(TextField)) />
    </Field>
  </MuiPickersUtilsProvider>
  <div />
</div>
`;
